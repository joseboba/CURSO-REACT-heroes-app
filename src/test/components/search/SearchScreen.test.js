import React from 'react';
import { SearchScreen } from '../../../components/search/SearchScreen';
import { MemoryRouter, Route } from 'react-router-dom';
import { mount } from 'enzyme';
import '@testing-library/jest-dom';

describe('Pruebas en el SearchScreen', () => {

    test('debe de mostrarse correctamente con valores por defecto ', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search']}>
                <Route path="/search" component={ SearchScreen } />
            </MemoryRouter>
        )

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.alert-info').text().trim()).toBe('Search a hero');
    })

    test('debe de mostrar a Batman y el input con el valor del query string', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Route path="/search" component={ SearchScreen } />
            </MemoryRouter>
        )

        expect(wrapper.find('input').prop('value')).toBe('batman');
    })

    test('debe de mostrar un error si no se encuentra el Hero', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman3445']}>
                <Route path="/search" component={ SearchScreen } />
            </MemoryRouter>
        )
        
        expect(wrapper.find('#error').text().trim()).toBe('Theres is not a hero with batman3445')
    })
    
    test('debe de mostrar el push del history', () => {
        
        const historyMock = {
            push: jest.fn()
        }

        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Route path="/search" component={ () => <SearchScreen history={ historyMock } /> } />
            </MemoryRouter>
        )

        wrapper.find('input').simulate('change', {
            target: {
                name: 'searchText',
                value: 'batman'
            }
        })
        wrapper.find('form').simulate('submit', () => preventDefault())

        expect(historyMock.push).toHaveBeenCalledWith('?q=batman')
    })
    
    
    

})
