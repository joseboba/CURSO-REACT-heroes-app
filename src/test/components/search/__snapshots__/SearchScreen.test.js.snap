// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el SearchScreen debe de mostrarse correctamente con valores por defecto  1`] = `
<div>
  <h1>
     Search Screen 
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <h4>
         Search Form 
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          onClick={[Function]}
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-8"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
